src/Compilers/CSharp/Portable/Symbols/ConstraintsHelper.cs

```diff
@@ -1201,7 +1200,7 @@ private static bool RequiresChecking(NamedTypeSymbol type)
             return true;
```

Comment:
I'm not confident this assertion is sound. If you have a generic `A<T>.B<U>`, and you construct `A<int>` and look up `B` in it, does `B` satisfy this assertion? I think the answer would be "no", but that cannot arise in correct source code. Can it arise in incorrect source code?